docs/building-apps/moneygram-access-integration-guide.mdx

```diff
@@ -185,10 +216,27 @@ You will need the following pieces of information:
 - The amount the user would like to withdraw / cash-out
   - This should be collected from the user prior to initiating this transaction
 
-The following code can be used as a reference for implementing this logic yourself. This code is not necessarily production-ready.
+The following code can be used as a reference for implementing this logic yourself.
 
 <CodeExample>
 
+```ts
+// Use same "anchor" object from previous step.
+const { url, id } = await anchor.sep24().withdraw({
+  authToken: authToken, // Use same "authToken" string from previous step
+  withdrawalAccount: FUNDS_STELLAR_KEYPAIR.public_key,
+  assetCode: ASSET_CODE, // Should be "USDC"
+  lang: "en",
+  extraFields: {
+    amount: "<amount to withdraw / cash-out>",
+  },
+});
+
+// Append the callback param for the wallet to be notified when the user has
+// completed the MGI experience and has requested to close the window.
+const urlWithCallback = `${url}&callback=postmessage`;
```

Comment:
We advise against using postMessage approach, I think we should mention that it's being deprecated long term

Reply:
+1 on this, do we even need to recommend it? I thought MoneyGram supported not using postmessage as well.

Reply:
I believe the problem of not implementing the `postMessage` handler is that MGI is still displaying a red `OK/Close` button on the last screen. So if wallets don't listen for that button it'll behave like a broken button.

But one thing I just realized is that Vibrant is not appending this `&callback=postmessage` param to the url but the app still receives the `postMessage` event. So it looks to me that we don't need to ask wallets to append this param, but instead just listen to a postMessage event coming from that last red button so they know it's safe to close the webview and prevent the button from looking broken. What do you guys think?

Reply:
Hmm so if we don't listen to it, the button will not work? I think we should describe it in the documentation then

Reply:
We defined a migration plan with MGI to remove postmessages because of the UX complications in the event that something doesn't work. @Ifropc can you dig up the plans we had and revisit this with MGI? Feel free to loop in the Vibrant team as well.

Reply:
Thanks guys, would be nice to remove that button.

Just to clarify: Vibrant does not rely on the postMessage event anymore for a while now, users can close the webview without tapping MGI's red button. Which means MGI would just need to remove that red button and probably add some copy letting users know they can safely close the window.

Reply:
@JakeUrban they have been stalling the plan for a very long time, it's somewhere in their backlog. We should raise this question again with their team

Reply:
Yep the idea is to replace button with the next that says you can close the window now

Reply:
https://github.com/stellar/stellar-docs/pull/658/commits/960e7cc8617115e256fe08e280c5c72d3592a9de